go-controller/pkg/kubevirt/pod.go

```diff
@@ -57,3 +57,24 @@ func FindNetworkInfo(client *factory.WatchFactory, pod *corev1.Pod) (NetworkInfo
 	}
 	return networkInfo, nil
 }
+
+// PodIsLiveMigrationLeftOver return true if there are other pods related to
+// to it and any of them has newer creation timestamp.
+func PodIsLiveMigrationLeftOver(client *factory.WatchFactory, pod *corev1.Pod) (bool, error) {
+	vmPods, err := FindVMRelatedPods(client, pod)
+	if err != nil {
+		return false, fmt.Errorf("failed finding related pods for pod %s/%s when checking live migration left overs: %v", pod.Namespace, pod.Name, err)
+	}
+
+	for _, vmPod := range vmPods {
+		if vmPod.CreationTimestamp.After(pod.CreationTimestamp.Time) {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func PodMatchesExternalIDs(pod *corev1.Pod, externalIDs map[string]string) bool {
```

Comment:
this function name is IMO misleading. 

You're not checking if the external IDs match; you're checking if **particular** things in the external IDs match. Can you come up with a better naming ? i.e. what does this represent ? Maybe `isPodFromSameVM` ?

Just trying to say this function's name focuses on the "how" something is done instead of what it is doing.

Reply:
This checks that a nbdb resource is own by the VM backed up by the pod,  maybe `kubevirt.PodOwnsNbResource`